src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -0,0 +1,539 @@
+                // In this case, we need to iterate through the <TargetFrameworks>, set <TargetFramework> with
+                // each value, and build the project.
+
+                var hasTargetFrameworkProp = _loadedProject.GetProperty(PropertyNames.TargetFramework) != null;
+                var targetFrameworks = targetFrameworksValue.Split(';');
+                var results = ImmutableArray.CreateBuilder<ProjectFileInfo>(targetFrameworks.Length);
+
+                foreach (var targetFramework in targetFrameworks)
+                {
+                    _loadedProject.SetProperty(PropertyNames.TargetFramework, targetFramework);
+                    _loadedProject.ReevaluateIfNecessary();
+
+                    var projectFileInfo = await BuildProjectFileInfoAsync(cancellationToken).ConfigureAwait(false);
+
+                    results.Add(projectFileInfo);
+                }
+
+                // Remove the <TargetFramework> property if it didn't exist in the file before we set it.
+                // Otherwise, set it back to it's original value.
+                if (!hasTargetFrameworkProp)
+                {
+                    var targetFrameworkProp = _loadedProject.GetProperty(PropertyNames.TargetFramework);
+                    _loadedProject.RemoveProperty(targetFrameworkProp);
+                }
+                else
+                {
+                    _loadedProject.SetProperty(PropertyNames.TargetFramework, targetFrameworkValue);
+                }
+
+                _loadedProject.ReevaluateIfNecessary();
+
+                return results.ToImmutable();
+            }
+            else
+            {
+                var projectFileInfo = await BuildProjectFileInfoAsync(cancellationToken).ConfigureAwait(false);
+
+                return ImmutableArray.Create(projectFileInfo);
+            }
+        }
+
+        private async Task<ProjectFileInfo> BuildProjectFileInfoAsync(CancellationToken cancellationToken)
+        {
+            var project = await _buildManager.BuildProjectAsync(_loadedProject, Log, cancellationToken).ConfigureAwait(false);
+
+            return project != null
+                ? CreateProjectFileInfo(project)
+                : ProjectFileInfo.CreateEmpty(Language, _loadedProject.FullPath, Log);
+        }
+
+        private ProjectFileInfo CreateProjectFileInfo(MSB.Execution.ProjectInstance project)
+        {
+            var commandLineArgs = GetCommandLineArgs(project);
+
+            var outputFilePath = project.ReadPropertyString(PropertyNames.TargetPath);
+            if (!string.IsNullOrWhiteSpace(outputFilePath))
+            {
+                outputFilePath = GetAbsolutePathRelativeToProject(outputFilePath);
+            }
+
+            var outputRefFilePath = project.ReadPropertyString(PropertyNames.TargetRefPath);
+            if (!string.IsNullOrWhiteSpace(outputRefFilePath))
+            {
+                outputRefFilePath = GetAbsolutePathRelativeToProject(outputRefFilePath);
```

Comment:
Sorry, I didn't follow that.

Reply:
I _think_ you misunderstood `ReadPropertyString()`. This does not just trigger the property's existence, but also returns the value if it exists. So, if the user has an empty `<TargetRefPath></TargetRefPath>`, this will still assume that an `OutputRefPath` was not specified. However, if there _was_ something meaningful in this property, we ensure that the path is relative to the project.